src/cargo/ops/cargo_compile/unit_generator.rs

```diff
@@ -475,6 +482,22 @@ impl<'a> UnitGenerator<'a, '_> {
         let mut scrape_proposals = self.filter_targets(can_scrape, false, CompileMode::Docscrape);
         scrape_proposals.retain(|proposal| pkgs_to_scrape.contains(proposal.pkg));
 
+        let skipped_examples = skipped_examples.into_inner();
+        if !skipped_examples.is_empty() {
+            let mut shell = self.ws.config().shell();
+            let example_str = skipped_examples
+                .into_iter()
+                .map(|t| t.description_named())
+                .collect::<Vec<_>>()
+                .join(", ");
+            shell.warn(format!(
+                "\
+Rustdoc did not scrape the following examples because they require dev-dependencies: {example_str}
+    If you want Rustdoc to scrape these examples, then add `doc-scrape-examples = true`
```

Comment:
```suggestion
            let examples = skipped_examples.join(", ");
            shell.warn(format!(
                "\
the following examples are not scraped because they require dev-dependencies: {examples}
    If you want to scrape these examples, then add `doc-scrape-examples = true`
```